src/main/java/com/w6/controller/EndpointController.java

```diff
@@ -120,9 +122,24 @@ public ModelAndView update(
     }
     
     @RequestMapping(value = "/input", method = RequestMethod.GET)
-    public String displayInput() 
+    public ModelAndView displayInput(@RequestParam(value = "emailid", required = false) Long emailId) throws IOException 
     {
-        return INPUT_VIEW;
+        ModelAndView modelAndView = new ModelAndView(INPUT_VIEW);
+        
+        try {
+            if (emailId != null) 
+            {
```

Comment:
Mikhail use this style =) He advised me to use this style. I don't like it as well